src/ui/Analytics/Analytics.ts

```diff
@@ -463,6 +469,14 @@ export class Analytics extends Component {
     );
   }
 
+  private handleCoveoUAEventReady(data: ICoveoUAEventArgs) {
+    if (this.options.populateDataLayer) {
```

Comment:
I would invert the conditions to have more defensive and less nested control-flow. e.g.
```
private handleCoveoUAEventReady(data: ICoveoUAEventArgs) {
    if (!this.options.populateDataLayer) {
        return;
    }

    const dataLayerName = this.options.dataLayerName;

    if (!<any>window)[dataLayerName]) {
        (<any>window)[dataLayerName] = []
    }

    (<any>window)[dataLayerName].push(data);
}
```

Reply:
👍